airframe-sql/src/test/scala/wvlet/airframe/sql/analyzer/TypeResolverTest.scala

```diff
@@ -350,10 +350,17 @@ class TypeResolverTest extends AirSpec {
     test("resolve having") {
       val p = analyze("select id, count(*) cnt from A group by id having cnt > 10")
       p shouldMatch {
-        case Aggregate(_, _, List(GroupingKey(`ra1`, _)), Some(GreaterThan(col, LongLiteral(10, _), _)), _) =>
-          col shouldMatch {
-            case FunctionCall("count", Seq(AllColumns(None, Some(Seq(`ra1`, `ra2`)), _)), false, None, None, _) => ()
+        case Aggregate(
+              _,
+              List(c1, c2 @ SingleColumn(f: FunctionCall, _, _, _)),
+              List(GroupingKey(`ra1`, _)),
+              Some(GreaterThan(col, LongLiteral(10, _), _)),
+              _
+            ) if c1.name == "id" && c2.name == "cnt" && f.functionName == "count" =>
+          f.args shouldMatch { case List(AllColumns(_, Some(cols), _)) =>
+            cols.toSet shouldBe Set(ra1, ra2)
           }
+          col shouldMatch { case ResolvedAttribute("cnt", DataType.LongType, _, sourceColumns, _) => }
```

Comment:
I think the sourceColumns need to be the all of the column data used for generating this ResolvedAttribute. If we use airframe-sql for tracking data lineage, we may need to maintain this property across SQL expression nodes.

Reply:
For example, if the function is max/min/max_by/min_by etc., source column information will be necessary to show the result contains any PID. If the function is count(1), personal information will be lost, so sourceColumns might be unnecessary because raw column values will not be exposed.

Reply:
`sourceColumns` seems an empty List in this pull request. Is this expected?

Reply:
How about defining `sourceColumns` as input columns without any modifications so that data lineage with nested function calls, such as `count(max(..(col)))`, can be traced in a different manner outside TypeResolver? 

If it sounds good, I will fix code and tests accordingly.

Reply:
Hmm, I think `TypeResolver` will be unnecessary in that case. Just scanning a tree for any node would work for extracting column lineage.

Reply:
ok. Let's forget about data lineage. It's not relevant in this PR.

At least, we need to be able to track whether PID/private columns are used as function inputs. 

```scala
          col shouldMatch { case ResolvedAttribute("cnt", DataType.LongType, _, sourceColumns, _) => }
```
this souceColumn should have Set(ra1, ra2)

Reply:
We need to give resolved `selectItems` instead of `a.outputAttributes` (= `selectItems`) here.
https://github.com/wvlet/airframe/blob/df40e3f1cc309df963cac8f996f3d2b5b1e885e4/airframe-sql/src/main/scala/wvlet/airframe/sql/analyzer/TypeResolver.scala#L118-L122

Reply:
By the way, I noticed that this query is actually invalid on Presto:
```sql
select id, count(*) cnt from A group by id having cnt > 10
```
Got the following error:
```
Query 20221219_122609_00005_r3q5y failed: line 1:84: Column 'cnt' cannot be resolved
```

Reply:
Good catch 😅

Reply:
- Fixed https://github.com/wvlet/airframe/pull/2649#discussion_r1052558235
- Make explicit that ResolvedAttribute.sourceColumns is given only when the attribute is a direct reference of table columns without any change (e.g., via function).

Reply:
> Make explicit that ResolvedAttribute.sourceColumns is given only when the attribute is a direct reference of table columns without any change

Oh, I see.